packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -77,7 +77,10 @@ void copyDirectorySync(Directory srcDir, Directory destDir) {
     destDir.createSync(recursive: true);
 
   srcDir.listSync().forEach((FileSystemEntity entity) {
-    String newPath = fs.path.join(destDir.path, fs.path.basename(entity.path));
+    String basename = destDir.fileSystem.path.joinAll(
+        entity.fileSystem.path.split(entity.fileSystem.path.basename(entity.path))
+    );
+    String newPath = fs.path.join(destDir.path, basename);
```

Comment:
Using `fs` here assumes that `destDir.fileSystem == fs`, no?  Instead, we should just explicitly use `destDir.fileSystem`.